api/src/devcomp/domain/services/solution-service-qrocm-ind.js

```diff
@@ -37,40 +37,56 @@ function match({ answerValue, solution }) {
 }
 
 function _applyTreatmentsToSolutions(solutions, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(solutions, (solution, solutionKey) => {
-    solution.forEach((variant, variantIndex) => {
-      if (qrocBlocksTypes[solutionKey] === 'select') {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, []);
-      } else {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, enabledTreatments);
+  const treatedSolutions = {};
+  for (const solutionKey in solutions) {
+    const solutionVariants = solutions[solutionKey];
+    const solutionType = qrocBlocksTypes[solutionKey];
+
+    treatedSolutions[solutionKey] = solutionVariants.map((variant) => {
+      switch (solutionType) {
+        case 'select':
+          return applyTreatments(variant, []);
+        case 'text':
+        default:
+          return applyTreatments(variant, enabledTreatments);
```

Comment:
Un peu embêté d'utiliser un switch pour un seul cas. 

```suggestion
if (solutionType === 'select')
	return applyTreatments(variant, []);
}
return applyTreatments(variant, enabledTreatments);
```

Reply:
ok pour repartir sur un if else classique, + simple à lire.